### ISSUE TITLE
milestone 6: bake-off — tune Qwen vs SEA-LION, publish the table, pick

### ISSUE THREAD

**Issue by opariffazman** [officialdad/camne#2]
## Goal

The bake-off is run and published: both candidate bases LoRA-tuned on the milestone-5 dataset, both evaluated on the BM-register InterCode-ALFA 300 + English control 300, README table filled, winner picked on numbers (accuracy, perf on the target box, licence). The winning GGUF is pinned in `internal/provision` and shipped.

## Spec

[PROMPT.md §5 (bake-off + training recipe), §7 (benchmark protocol)](../blob/main/PROMPT.md). CLAUDE.md "Model work — measure or it did not happen" is binding: no merge without the table.

## State when this issue was written

Repo at `fe0c247`, v0.2.0 shipped with whatisit's English nl2sh-1.5b (a Qwen2.5-Coder-1.5B tune) — that model IS the untuned-base comparison row for the Qwen arm. Engine already matches benchmark settings (temperature 0, n_predict 64, GBNF single line — `internal/engine/engine.go`). Model swap = three consts + size in `internal/provision/provision.go`.

**Blocked by:** milestone 5 dataset (see the milestone 5 issue).

Decision from discussion (2026-08-13): training machine is an **RTX 3090 (24 GB)**, not the A100 the recipe assumed. Confirmed workable: keep effective batch 32 (drop micro-batch, raise grad accum if SEA-LION OOMs; add gradient checkpointing), bf16 is native on Ampere, everything else identical (r32/α64/dropout 0.05, all-linear, 2e-4 cosine 3% warmup, 2 epochs, seq 512, seed 42). Expect 3–6 h per run instead of ~1 h. Do NOT switch to QLoRA — quantized base = a second changed variable.

## Machine / resource requirements

- Training: RTX 3090 24 GB (or better), ~50 GB disk, overnight for both runs.
- Eval: the **target box** (4 cores / 8 GB / no GPU) for perf numbers; any Docker-capable machine for InterCode-ALFA accuracy runs (scorer runs commands in containers).
- HF account for base downloads + publishing tuned weights.

## Deliverables

- [ ] LoRA runs: Qwen2.5-Coder-1.5B-Instruct and Gemma-SEA-LION-v4.5-E2B-IT, same dataset, same recipe, seed 42 (mallam-1.1b third seat only if both disappoint)
- [ ] Merged bf16 → f16 GGUF → Q4_K_M for each
- [ ] Accuracy: InterCode-ALFA **unmodified scorer**, BM-colloquial 300 + English control 300, temperature 0, max_tokens 64, threshold 0.75; paired exact McNemar vs each untuned base with 95% CI (report "cannot resolve" if 300 tasks cannot)
- [ ] Perf on the target box: tok/s (median ≥ 12 queries), warm latency, cold start, RSS, disk, install wall time; thread sweep 2/4/6/8
- [ ] README table in the PROMPT.md §7.3 shape, untuned bases as rows, licence column (SEA-LION inherits Gemma terms — record the licence cost in the table, not a footnote)
- [ ] Winner pinned in `internal/provision/provision.go`, release cut

## How to resume

```sh
git clone https://github.com/officialdad/camne && cd camne
go build ./... && go test ./...
# Confirm milestone 5 dataset is published first (blocking).
# Training env is Python (peft/transformers) — lives OUTSIDE this repo's
# Go-only rule; keep training scripts in a training/ dir or separate repo,
# record the exact recipe + seed in the PR either way.
# Start from whatisit's validated recipe: 0.540 → 0.620, p = 0.004.
```

## References

- https://huggingface.co/Qwen/Qwen2.5-Coder-1.5B-Instruct (Apache-2.0)
- https://huggingface.co/aisingapore/Gemma-SEA-LION-v4.5-E2B-IT-GGUF (Gemma terms)
- https://huggingface.co/mesolitica/mallam-1.1b-20k-instructions (third seat)
- https://github.com/westenfelder/InterCode-ALFA — scorer, run unmodified
- https://github.com/ThorOdinson246/whatisit-nl2sh — validated recipe + methodology

**Comment by opariffazman** [comment]
Blocked by #1 (dataset).

**Comment by opariffazman** [comment]
Done. Both arms tuned on the same pool with the same recipe; table and reasoning in RESULTS.md ("The SEA-LION arm"). SEA-LION lost every register and fails constraint 3 (4.18 GB RSS); Qwen2.5-Coder-1.5B tune shipped in v0.7.0 and superseded by run 7 in v0.8.0. Licence cost is in RESULTS.md prose rather than a README table column.
